### ISSUE TITLE
Installing azure-cli takes a really long time on Ubuntu (18.04)

### ISSUE THREAD

**Issue by aditimalladi** [Azure/azure-cli#18531]
**Describe the bug**
Installing azure-cli on 18.04 Ubuntu is intermittently **very** slow (currently taking over an hour to install hence resulting in timeouts.)

**To Reproduce**
Running command:  `curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash`

**Expected behavior**
Expected to install az cli in a within a time span much lesser than 60 minutes

**Environment summary**
Ubuntu 18.04

**Additional context**
Not seeing any explicit errors on trying to install, just waiting on install to complete: 
```
The following NEW packages will be installed:
  azure-cli
0 upgraded, 1 newly installed, 0 to remove and 21 not upgraded.
Need to get 62.4 MB of archives.
After this operation, 923 MB of additional disk space will be used.
Get:1 https://packages.microsoft.com/repos/azure-cli bionic/main amd64 azure-cli all 2.25.0-1~bionic [62.4 MB]
```

**Comment by kevinnesta** [comment]
it's is becoming a big problem for us too.
is it related to the 'resolved' outage https://status.dev.azure.com/_event/246136621 ?

**Comment by gregjhogan** [comment]
> is it related to the 'resolved' outage https://status.dev.azure.com/_event/246136621 ?

I think that event was the issue https://github.com/Azure/azure-cli/issues/18514 that occurred yesterday, today there is a new issue

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by jiasli** [comment]
I am able to reproduce the slow download speed from Microsoft CorpNet:

```
The following packages will be upgraded:
  azure-cli
1 upgraded, 0 newly installed, 0 to remove and 131 not upgraded.
Need to get 62.4 MB of archives.
After this operation, 12.0 MB of additional disk space will be used.
Get:1 https://packages.microsoft.com/repos/azure-cli bionic/main amd64 azure-cli all 2.25.0-1~bionic [62.4 MB]
2% [1 azure-cli 1261 kB/62.4 MB 2%]                                                                 54.5 kB/s 18min 42s
```

We are working internally with Microsoft APT Linux repos team for a solution.

**Comment by jiasli** [comment]
As tested, the downlead speed has recovered:

```
Get:1 https://packages.microsoft.com/repos/azure-cli bionic/main amd64 azure-cli all 2.25.0-1~bionic [62.4 MB]
Fetched 62.4 MB in 2s (28.6 MB/s)
```

Please try again and let us know if the issue is resolved.

**Comment by tankbob** [comment]
I'm still seeing the issue - building a docker image on DevOps pipeline, same build works with no problems on local machine

**Comment by sandersaares** [comment]
I am still seeing this issue in both local machine (North Europe region) and CXDPx pipelines (unknown region).

**Comment by nakazonot** [comment]
Things seem to change depending on the time of day.
Yesterday, the download speed was slow from 6:00 to 22:00 in UTC.

**Comment by jiasli** [comment]
The Linux repo team has scaled up the server of https://packages.microsoft.com/. If you are still seeing slow download speed, please let us know.

**Comment by jiasli** [comment]
We have worked with the Linux repo team to scale up the server of https://packages.microsoft.com/. If you are still seeing slow download speed, please let us know.

**Comment by DennisSc** [comment]
Hi,
at the moment, I am trying to download NET core 3.1 SDK from "packages.microsoft.com", using a 250MBit DSL line but only getting 40 KBytes/sec. download speed

**Comment by jiasli** [comment]
For .NET Core issue, please feel free to create a new issue at https://github.com/dotnet/core

**Comment by DennisSc** [comment]
I'd rather call it not a NET core issue, but a network bandwidth issue (affecting the same location that your team uses to host the azure-cli downlaods, mind you)

anyway it finished downloading, eventually.
"Fetched 80.9 MB in 12min 54s (104 kB/s)"

**Comment by dinvlad** [comment]
Same issue today, for some reason. I've build the same VS Code dev container just an hour ago and it was blazing fast. Now, on a rebuild the speed slowed down to what feels like 100x, trying to fetch packages from https://packages.microsoft.com/

**Comment by jiasli** [comment]
@dinvlad, I am not able to repro this issue currently:

```
$ curl https://packages.microsoft.com/repos/azure-cli/pool/main/a/azure-cli/azure-cli_2.30.0-1~bullseye_all.deb --remote-name
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 63.4M  100 63.4M    0     0  5983k      0  0:00:10  0:00:10 --:--:-- 5972k
```

Does it work if you try using some other network, like using Azure VM or mobile phone?

**Comment by dinvlad** [comment]
@jiasli - it was very intermittent. I tried later yesterday and it was working fine..

**Comment by hokb** [comment]
currently waiting on ~40kb/s DL rate, Hamburg, Germany

**Comment by NoahStahl** [comment]
Currently 300 kbps download of az cli on a 1 gb fiber connection.

**Comment by sdebruyn** [comment]
We are seeing consistent 100 kbps download speeds for over the last few months from multiple Belgian IPs and ISPs. Connection is capable of 1 Gbps

**Comment by jiasli** [comment]
@NoahStahl, could you let me know your location?

I am reporting this issue to Microsoft Linux Repo team for investigation.

**Comment by daviddavis** [comment]
I've filed an issue to investigate the slow download speeds: https://github.com/microsoft/linux-package-repositories/issues/13

**Comment by NoahStahl** [comment]
In my case, downloading in Austin TX USA

**Comment by chrisdoman** [comment]
For those experiencing crazy slow build processes because of this - I ended up just doing this as a workaround for now:

curl https://(my bucket).s3.amazonaws.com/azure-cli_2.38.0-1_bullseye_all.deb	 --output /tmp/azure.deb
dpkg -i /tmp/azure.deb
